internal/cmd/controller/target/target.go

```diff
@@ -193,15 +205,13 @@ func (m *Manager) BundlesForCluster(cluster *fleet.Cluster) (bundlesToRefresh, b
 }
 
 func (m *Manager) GetBundleDeploymentsForBundleInCluster(bundle *fleet.Bundle, cluster *fleet.Cluster) (result []*fleet.BundleDeployment, err error) {
```

Comment:
In case these changes were not enough, I wonder if applying similar changes eg. [here](https://github.com/rancher/fleet/blob/master/internal/cmd/controller/controllers/git/git.go#L525) could help reduce CPU load.

Reply:
Nice point Tan. That code surely benefit from using a similar approach, although it would need a different index, based on `GitRepo` instead of `Bundles`. I can create a new issue for that, and add it separately. Do you agree?

Reply:
Of course, this can be handled via a separate PR. I think it is relevant to the same issue as this one (#1842), but no strong opinions here.